lib/runnable.js

```diff
@@ -116,9 +116,14 @@ Runnable.prototype.enableTimeouts = function(enabled) {
  * @memberof Mocha.Runnable
  * @public
  * @api public
+ * @param {string} reason - Reason the Runnable is being skipped.
  */
-Runnable.prototype.skip = function() {
-  throw new Pending('sync skip');
+Runnable.prototype.skip = function(reason) {
+  if (arguments.length > 0) {
+    reason = String(reason);
+    this.reason = reason;
+  }
```

Comment:
Why just check `if (reason) {`?

In a case of`it.skip()`, `arguments` is `{ '0': undefined }` and `arguments.length` is `1`, because of `this.runnable().skip(reason)` in `lib/context.js`. 
Therefore, `reason` is `'undefiled'` as string and it print like thi:
```
- should skip immediately with reason (undefined)
```

Reply:
Ah, good catch. Originally I has `if (reason) {`, but thought this would be better for some reason. I'll change it back.

Reply:
I also considered using `utils.isString` to validate the input, but wasn't sure if that was needed here.